include/SimCore/GammaPhysics.h

```diff
@@ -38,42 +41,34 @@ class GammaPhysics : public G4VPhysicsConstructor {
    * @param parameters The python configuration
    */
   GammaPhysics(const G4String& name,
-               const framework::config::Parameters& parameters)
-      : G4VPhysicsConstructor(name),
-        modelParameters{parameters.getParameter<framework::config::Parameters>(
-            "photonuclear_model")} {}
-  GammaPhysics(const G4String& name = "GammaPhysics");
+               const framework::config::Parameters& parameters);
 
   /**
    * Class destructor.
    */
   virtual ~GammaPhysics() = default;
 
   /**
-   * Construct particles (no-op).
+   * Construct particles
+   *
+   * We don't do anything here since we are just attaching/updating
+   * the photon physics.
    */
-  void ConstructParticle() {}
+  void ConstructParticle() final override;
```

Comment:
Nice catch, although having both final and override is a bit overkill, final implies override (see https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c128-virtual-functions-should-specify-exactly-one-of-virtual-override-or-final)